driver/src/main/java/org/neo4j/driver/internal/net/SocketClient.java

```diff
@@ -120,7 +119,10 @@ public void start()
         try
         {
             logger.debug( "~~ [CONNECT] %s", address );
-            setChannel( ChannelFactory.create( address, securityPlan, timeoutMillis, logger ) );
+            if( channel == null )
```

Comment:
Could we instead assert that channel is `null` as the first thing in this method and throw `IllegalStateException` if it is not null. This will disallow multiple `#start()` calls.